Using anstream with tracing_appender's `RollingFileAppender`

anstream's `StripStream` requires `S: AsLockedWrite` to implement `std::io::Write`. `AsLockedWrite` is implemented for `Box<dyn std::io::Write>` and tracing_appender's `RollingFileAppender` implements `std::io::Write`, so it looks like what is needed is just a `Box`. However, it doesn't actually work:
```rust
let file_appender = tracing_appender::rolling::daily(logs, "prefix");
let stream = Box::new(file_appender) as Box<dyn std::io::Write + Send>;
let strip_stream = anstream::StripStream::new(stream);
let (non_blocking, guard) = tracing_appender::non_blocking(strip_stream);
```
```rust
error[E0277]: the trait bound `Box<dyn std::io::Write + Send>: AsLockedWrite` is not satisfied
   --> src\main.rs:26:72
    |
26  |             let (non_blocking, guard) = tracing_appender::non_blocking(strip_stream);
    |                                         ------------------------------ ^^^^^^^^^^^^ the trait `AsLockedWrite` is not implemented for `Box<dyn std::io::Write + Send>`, which is required by `StripStream<Box<dyn std::io::Write + Send>>: std::io::Write`
    |                                         |
    |                                         required by a bound introduced by this call
    |
    = help: the trait `AsLockedWrite` is implemented for `Box<dyn std::io::Write>`
    = note: required for `StripStream<Box<dyn std::io::Write + Send>>` to implement `std::io::Write`
note: required by a bound in `non_blocking`
   --> C:\Users\Chaoses\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tracing-appender-0.2.3\src\lib.rs:193:24
    |
193 | pub fn non_blocking<T: Write + Send + 'static>(writer: T) -> (NonBlocking, WorkerGuard) {
    |                        ^^^^^ required by this bound in `non_blocking`

For more information about this error, try `rustc --explain E0277`.
```
This error message doesn't make sense since `stream` already satisfies `Box<dyn std::io::Write + Send>`. And if I remove `Send`, there will be another error:
```rust
error[E0277]: `dyn std::io::Write` cannot be sent between threads safely
   --> src\main.rs:26:72
    |
26  |             let (non_blocking, guard) = tracing_appender::non_blocking(strip_stream);
    |                                         ------------------------------ ^^^^^^^^^^^^ `dyn std::io::Write` cannot be sent between threads safely
    |                                         |
    |                                         required by a bound introduced by this call
    |
    = help: the trait `Send` is not implemented for `dyn std::io::Write`, which is required by `StripStream<Box<dyn std::io::Write>>: Send`
    = note: required for `Unique<dyn std::io::Write>` to implement `Send`
note: required because it appears within the type `Box<dyn std::io::Write>`
   --> C:\Users\Chaoses\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\alloc\src\boxed.rs:234:12
    |
234 | pub struct Box<
    |            ^^^
note: required because it appears within the type `StripStream<Box<dyn std::io::Write>>`
   --> C:\Users\Chaoses\.cargo\registry\src\index.crates.io-6f17d22bba15001f\anstream-0.6.15\src\strip.rs:7:12
    |
7   | pub struct StripStream<S>
    |            ^^^^^^^^^^^
note: required by a bound in `non_blocking`
   --> C:\Users\Chaoses\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tracing-appender-0.2.3\src\lib.rs:193:32
    |
193 | pub fn non_blocking<T: Write + Send + 'static>(writer: T) -> (NonBlocking, WorkerGuard) {
    |                                ^^^^ required by this bound in `non_blocking`

For more information about this error, try `rustc --explain E0277`.
```

Any idea about what's going wrong?

--- a/crates/anstream/src/stream.rs
+++ b/crates/anstream/src/stream.rs
@@ -23,6 +23,8 @@ impl RawStream for std::io::Stderr {}
 impl RawStream for std::io::StderrLock<'_> {}
 
 impl RawStream for dyn std::io::Write {}
+impl RawStream for dyn std::io::Write + Send {}
+impl RawStream for dyn std::io::Write + Send + Sync {}
 
 impl RawStream for Vec<u8> {}
 
@@ -87,6 +89,20 @@ impl IsTerminal for std::io::StderrLock<'_> {
 }
 
 impl IsTerminal for dyn std::io::Write {
+    #[inline]
+    fn is_terminal(&self) -> bool {
+        false
+    }
+}
+
+impl IsTerminal for dyn std::io::Write + Send {
+    #[inline]
+    fn is_terminal(&self) -> bool {
+        false
+    }
+}
+
+impl IsTerminal for dyn std::io::Write + Send + Sync {
     #[inline]
     fn is_terminal(&self) -> bool {
         false
@@ -215,6 +231,8 @@ mod private {
     impl Sealed for std::io::StderrLock<'_> {}
 
     impl Sealed for dyn std::io::Write {}
+    impl Sealed for dyn std::io::Write + Send {}
+    impl Sealed for dyn std::io::Write + Send + Sync {}
 
     impl Sealed for Vec<u8> {}
 

--- a/crates/anstyle-wincon/src/stream.rs
+++ b/crates/anstyle-wincon/src/stream.rs
@@ -42,6 +42,28 @@ impl WinconStream for dyn std::io::Write {
     }
 }
 
+impl WinconStream for dyn std::io::Write + Send {
+    fn write_colored(
+        &mut self,
+        fg: Option<anstyle::AnsiColor>,
+        bg: Option<anstyle::AnsiColor>,
+        data: &[u8],
+    ) -> std::io::Result<usize> {
+        crate::ansi::write_colored(self, fg, bg, data)
+    }
+}
+
+impl WinconStream for dyn std::io::Write + Send + Sync {
+    fn write_colored(
+        &mut self,
+        fg: Option<anstyle::AnsiColor>,
+        bg: Option<anstyle::AnsiColor>,
+        data: &[u8],
+    ) -> std::io::Result<usize> {
+        crate::ansi::write_colored(self, fg, bg, data)
+    }
+}
+
 impl WinconStream for std::fs::File {
     fn write_colored(
         &mut self,

